api/environment.go

```diff
@@ -0,0 +1,148 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aau-network-security/haaukins/lab"
+	"github.com/aau-network-security/haaukins/store"
+	"github.com/aau-network-security/haaukins/svcs/guacamole"
+	"github.com/aau-network-security/haaukins/virtual/docker"
+	"github.com/rs/zerolog/log"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type environment struct {
+	timer		*time.Time
+	challenges 	[]store.Tag
+	lab 		lab.Lab
+	guacamole 	guacamole.Guacamole
+	guacPort 	uint
+}
+
+func (e environment) Assign(client *ClientRequest) error {
```

Comment:
This seems like a re-use of existing code, right? 
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself